### PR TITLE
Add punch-through sdc API

### LINKED ISSUE #323: SDC punch-through API
API to pass through a SDC blackbox through `vlsi.inputs`.

### PR DESCRIPTION
**Pull request by colinschmidt**
Fixes #323

### CODE REVIEW

File: src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py [hunk 1]
```diff
@@ -1018,6 +1018,11 @@ def sdc_pin_constraints(self) -> str:
                 direction=delay.direction,
                 name=delay.name
             ))
+
+        # Custom pin constraints that are verbatim appended
```
Comment (jwright6323) [inline src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py]:
```suggestion
        # Custom sdc constraints that are verbatim appended
```
Also `verbatim appended` sounds off to me. `appended verbatim` sounds better but I don't particularly care.

File: src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py [hunk 2]
```diff
@@ -1018,6 +1018,11 @@ def sdc_pin_constraints(self) -> str:
...
+        for custom in self.get_setting("vlsi.inputs.custom_sdc_constraints"):
```
Comment (edwardcwang) [inline src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py]:
```suggestion
        custom_sdc_constraints = self.get_setting("vlsi.inputs.custom_sdc_constraints")  # type: List[str]
        for custom in custom_sdc_constraints:
```

**Comment by colinschmidt** [comment]
Stepping afk for a bit feel free to press the button whenever.

**Comment by edwardcwang** [comment]
Actually hold for a short while; let me see if I can add a test

**Pull request merged**